### ISSUE TITLE
Game not saving

### ISSUE THREAD

**Issue by brandosha** [brandosha/gameplaycolor#1]

**Comment by ewancube** [comment]
My steps to produce this issue:
1. Played SMB Deluxe on your hosted fork (version before 3.0.4). Created a save file via game (not a state save). Exited app.
2. Started the app. Received notice saying 3.0.4 was available and I needed to restart app. Did so.
3. Restarted app. Logged out of Google Drive. Logged into Google Drive again.
4. Opened SMB Deluxe. Previous save loaded successfully.
5. Advanced one stage. Re-saved. Exited app.
6. Restarted app. Opened SMB Deluxe. Save from prior to 3.0.4 upgrade is loaded.

Basically a save file created prior to 3.0.4 and Google Drive log in/out can’t be overwritten in 3.0.4. BTW, deleting the app from home screen and re-adding via Safari allowed new saves to be created in SMB Deluxe using 3.0.4 but at the cost of my previous save files.

**Comment by brandosha** [comment]
The problem was caused by a merge conflict related to indexeddb primary keys that I resolved incorrectly. It caused two copies of the save to be added to the db and one was being read from and the *other* one was being written to. The only way to fix it was to clean out the db so I added a "Clear save data" button in the settings so that anyone having issues can clear the db without having to reinstall and sign in again.
